radicle-cli/src/git.rs

```diff
@@ -60,6 +62,45 @@ impl From<String> for Rev {
     }
 }
 
+#[derive(Error, Debug)]
+pub enum RemoteError {
+    #[error("url malformed: {0}")]
+    ParseUrl(#[from] transport::local::UrlError),
+    #[error("remote `url` not found")]
+    MissingUrl,
+    #[error("remote `name` not found")]
+    MissingName,
+}
+
+#[derive(Clone)]
+pub struct Remote<'a> {
+    pub(crate) name: String,
+    pub(crate) url: radicle::git::Url,
+    inner: git2::Remote<'a>,
+}
+
+impl Remote<'_> {
+    pub fn refspecs(&self) -> git2::Refspecs {
+        self.inner.refspecs()
+    }
+}
+
+impl<'a> TryFrom<git2::Remote<'a>> for Remote<'a> {
+    type Error = RemoteError;
+
+    fn try_from(value: git2::Remote<'a>) -> Result<Self, Self::Error> {
+        let url = value.url().map_or(Err(RemoteError::MissingUrl), |url| {
+            Ok(radicle::git::Url::from_str(url)?)
+        })?;
+        let name = value.name().map_or(Err(RemoteError::MissingName), Ok)?;
```

Comment:
This is also fine, but it's more common to use `ok_or`, because it foregoes the need for `Err` or `Ok` :). I'm surprised clippy doesn't warn against it tbh.
```suggestion
        let name = value.name().ok_or(RemoteError::MissingName)?;
```